Hypothesis's fake Random disagrees with true Random when sampling empty sequences

The following code doesn't work correctly when using Hypothesis's fake random number generator:

```python
from hypothesis import given, strategies as st

@given(rnd=st.randoms(use_true_random=False))
def test_sample_from_empty(rnd):
    assert rnd.sample([], 0) == []
```

It fails with:

```
E           hypothesis.errors.InvalidArgument: Cannot sample from a length-zero sequence.
E           Falsifying example: test_sample_from_empty(
E               rnd=HypothesisRandom(generated data),
E           )
```

But the following test passes:

```
from random import Random

def test_sample_from_empty():
    rnd = Random()
    assert rnd.sample([], 0) == []
```

The issue is that although you can't sample a non-zero number of elements from an empty sequence, it's perfectly valid to sample 0 elements from an empty sequence, and the normal Random class supports this fine but our emulation of it does not.

--- a/hypothesis-python/RELEASE.rst
+++ b/hypothesis-python/RELEASE.rst
@@ -1,0 +1,8 @@
+RELEASE_TYPE: patch
+
+When :func:`~hypothesis.strategies.randoms` was called with `use_true_randoms=False`,
+calling `sample` on it with an empty sequence and 0 elements would result in an error,
+when it should have returned an empty sequence to agree with the normal behaviour of
+`random.Random`. This fixes that discrepancy.
+
+Fixes :issue:`3765``

--- a/hypothesis-python/src/hypothesis/strategies/_internal/random.py
+++ b/hypothesis-python/src/hypothesis/strategies/_internal/random.py
@@ -296,14 +296,17 @@ class ArtificialRandom(HypothesisRandom):
                     f"Sample size {k} not in expected range 0 <= k <= {len(seq)}"
                 )
 
-            result = self.__data.draw(
-                lists(
-                    sampled_from(range(len(seq))),
-                    min_size=k,
-                    max_size=k,
-                    unique=True,
+            if k == 0:
+                result = []
+            else:
+                result = self.__data.draw(
+                    lists(
+                        sampled_from(range(len(seq))),
+                        min_size=k,
+                        max_size=k,
+                        unique=True,
+                    )
                 )
-            )
 
         elif method == "getrandbits":
             result = self.__data.draw_bits(kwargs["n"])

--- a/hypothesis-python/tests/cover/test_randoms.py
+++ b/hypothesis-python/tests/cover/test_randoms.py
@@ -381,3 +381,8 @@ def test_can_sample_from_large_subset(rnd):
     ys = rnd.sample(xs, n)
     assert set(ys).issubset(set(xs))
     assert len(ys) == len(set(ys)) == n
+
+
+@given(st.randoms(use_true_random=False))
+def test_can_draw_empty_from_empty_sequence(rnd):
+    assert rnd.sample([], 0) == []

